### PR TITLE
Fixes gunpowder and teslium grenades being completely nonfunctional.

### LINKED ISSUE #54986: Gunpowder grenades don't work
## Testmerges:
- #54804 
- #54702 
- #54632 

## Reproduction:
1. Fill two large beakers with gunpowder
2. Heat the beakers to 470&deg;K
3. Create a grenade using the two beakers.
4. Prime and throw the grenade.
5. Watch as the grenade disappears and there is a deafening lack of explosion

### LINKED ISSUE #62461: Teslium activation is broken.
## Reproduction:
I have put a beaker with 50u of Teslium and a beaker with 50u of Water into my custom grenade. I have made 3 such grenades in total. When the grenades activated, nothing happened all 3 times, only the electricity noise & particles appeared.

There should've been 2 Tesla bolts spawned every time one of these grenades were activated, as the amount of Teslium participating in the reaction was greater than 40u.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Gunpowder and teslium both have delayed effects when they react. Grenades both react their reagents in a temporary reagent holder that gets immediately deleted after reacting and tend to delete themselves. As a result neither gunpowder nor teslium could do anything as their one reference point vanished the instant they reacted.

This makes chemical genades not magically delete themselves on detonation. As a result, gunpowder and teslium both work. You can also recycle nondestructive chemical grenade casings if you feel like taking them apart.

Also makes large grenade casings with slime cores behave slightly less... O(n^2).

## Why It's Good For The Game
- Less issues on the tracker:
  - Fixes #62461.
  - Fixes #54986.
- Slightly more constancy across explosives.
- Large grenade casings are no longer O(n^2) with the number of slime cores they contain.

## Changelog

:cl:
fix: Chemical grenades no longer evaporate into thin air upon detonation. As a result, gunpowder and teslium now work with them. Also, you can recycle cleaner grenades if you want to bother.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chem_splash.dm [hunk 1]
```diff
@@ -4,75 +4,110 @@
...
-	splash_holder.my_atom = epicenter // For some reason this is setting my_atom to null, and causing runtime errors.
-	var/total_temp = 0
+	var/tmp_holder = null
+	if (isnull(holder))
+		tmp_holder = TRUE
+		holder = new /datum/reagents(total_reagents * threatscale)
+		holder.my_atom = epicenter
+	else
+		tmp_holder = FALSE
+		holder.maximum_volume += total_reagents * threatscale
```
Comment (AnturK) [inline code/modules/reagents/chem_splash.dm:41]: You might want to mention in the docstring this modification happens.

File: code/game/machinery/syndicatebomb.dm [hunk 2]
```diff
@@ -434,10 +438,6 @@
 
 	playsound(loc, 'sound/effects/bamf.ogg', 75, TRUE, 5)
 
-	if(loc && istype(loc, /obj/machinery/syndicatebomb/))
```
Comment (AnturK) [inline code/game/machinery/syndicatebomb.dm:437]: What's going on here ? Surely when it's done core should be gone ?

File: code/modules/reagents/chem_splash.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/modules/reagents/chem_splash.dm:41]: It's undone at the end of the proc... but I see what you mean. Currently people might think the size of the original holder limits the size of the mix. Sure, I can do that.

File: code/game/machinery/syndicatebomb.dm (hunk 2 above)
Comment (TemporalOroboros) [reply code/game/machinery/syndicatebomb.dm:437]: I'm currently running on the principal that the bomb core is basically just a really, really big chemical grenade and the casing is the wiring to make it go off/anchor it to the floor. I'm leaving it up to the explosion it creates to destroy the core and casing. If there's no explosion I don't see a reason that either should be destroyed.

File: code/modules/reagents/chem_splash.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/modules/reagents/chem_splash.dm:41]: I've put a note about it in the docstring.

File: code/game/machinery/syndicatebomb.dm [hunk 3]
```diff
@@ -386,6 +386,10 @@
 	var/temp_boost = 50
 	var/time_release = 0
 
+/obj/item/bombcore/chemical/Initialize(mapload)
+	. = ..()
+	create_reagents(1000)
```
Comment (Watermelon914) [inline code/game/machinery/syndicatebomb.dm:391]: make this a variable

Comment (TemporalOroboros) [reply code/game/machinery/syndicatebomb.dm:391]: Alright.

Comment (TemporalOroboros) [reply code/game/machinery/syndicatebomb.dm]: Done.

**Pull request merged**